src/RegisterPage.cpp

```diff
@@ -299,25 +320,93 @@ RegisterPage::onBackButtonClicked()
+                label->setText("");
+                label->hide();
+                return true;
+        } else {
+                label->show();
+                showError(label, msg);
+                return false;
+        }
+}
+
+bool
+RegisterPage::checkFields()
 {
         error_label_->setText("");
+        error_username_label_->setText("");
+        error_password_label_->setText("");
+        error_password_confirmation_label_->setText("");
+
+        error_username_label_->hide();
+        error_password_label_->hide();
+        error_password_confirmation_label_->hide();
+
+        password_confirmation_->setValid(true);
+        server_input_->setValid(true);
+
+        bool all_fields_good = true;
+        if (username_input_->isModified() &&
+            !checkOneField(error_username_label_,
+                           username_input_,
+                           tr("The username must not be empty, and must contain only the "
+                              "characters a-z, 0-9, ., _, =, -, and /."))) {
+                all_fields_good = false;
+        } else if (password_input_->isModified() &&
+                   !checkOneField(error_password_label_,
+                                  password_input_,
+                                  tr("Password is not long enough (min 8 chars)"))) {
+                all_fields_good = false;
+        } else if (password_confirmation_->isModified() &&
+                   password_input_->text() != password_confirmation_->text()) {
+                error_password_confirmation_label_->show();
+                showError(error_password_confirmation_label_, tr("Passwords don't match"));
+                password_confirmation_->setValid(false);
+                all_fields_good = false;
+        } else if (server_input_->isModified() &&
+                   (!server_input_->hasAcceptableInput() || server_input_->text().isEmpty())) {
+                showError(tr("Invalid server name"));
+                server_input_->setValid(false);
+                all_fields_good = false;
+        }
+        if (!username_input_->isModified() || !password_input_->isModified() ||
+            !password_confirmation_->isModified() || !server_input_->isModified()) {
+                all_fields_good = false;
+        }
+        return all_fields_good;
+}
 
-        if (!username_input_->hasAcceptableInput()) {
-                registerError(tr("Invalid username"));
-        } else if (!password_input_->hasAcceptableInput()) {
-                registerError(tr("Password is not long enough (min 8 chars)"));
-        } else if (password_input_->text() != password_confirmation_->text()) {
-                registerError(tr("Passwords don't match"));
-        } else if (!server_input_->hasAcceptableInput()) {
-                registerError(tr("Invalid server name"));
+void
+RegisterPage::onRegisterButtonClicked()
+{
+        if (!checkFields()) {
+                showError(error_label_, tr("Regisration Failed"));
```

Comment:
```suggestion
                showError(error_label_, tr("One or more fields have invalid inputs. Please correct those issues and try again."));
```

Reply:
Just "Regisration Failed" is confusing and misspelled.

Reply:
Yep:)